backend/src/sync.js

```diff
@@ -91,6 +91,33 @@ async function saveBlocks(blocksInfo) {
                     });
                   })
                 ),
+                models.AccessKey.bulkCreate(
+                  blockInfo.transactions
+                    .filter(tx =>
```

Comment:
You are filtering the transactions which do not have any AddKey, but still include the wrole transaction with all the irrelevant actions. Drop the filter, flatMap on the transaction (that is already the case), filter the actions (this part is missing), and map the filtered values to the proper shape.

Notes:

1. you don't need to check for a stringly-typed action since `AddKey` is always represented as an object (where the key is "AddKey")
2. match the permission exhaustively, and throw an error if there is unknown permission type, so we catch it earlier.